### ISSUE TITLE
Tag filter component - X button not keyboard accessible

### ISSUE THREAD

**Issue by marcyb1966** [carbon-design-system/carbon#3775]
## Environment

> Operating system - Mac

> Browser - Chrome

> Automated testing tool and ruleset - N/A

> Assistive technology used to verify - N/A

## Detailed description
When testing the keyboard accessibility of my UI that uses Tag filter, I was able to tab to my tag, but couldn't get the keyboard to actually click the "X" button.

> What version of the Carbon Design System are you using? - React

> What did you expect to happen?
Expected to be able to click the "cancel" (i.e. X) button using the keyboard.
> What happened instead? 
Was unable to click the button using just the keyboard.  Space bar, return, arrow keys all did nothing.
> What WCAG 2.1 checkpoint does the issue violate?
Keyboard accessibility
## Steps to reproduce the issue

1. Render a Tag component with "filter" field specified
2. Tab to Tag, cancel button will highlight
3. Try to "click" the button using just the keyboard.  Nothing works.

> Please create a reduced test case in CodeSandbox
>
> - Style and vanilla JS:
>   https://codesandbox.io/s/github/carbon-design-system/carbon/tree/master/packages/components/examples/codesandbox
> - React:
>   https://codesandbox.io/s/github/carbon-design-system/carbon/tree/master/packages/react/examples/codesandbox
Example can be seen here.  Clicking with the mouse will show an alert, but if you try to click using just the keyboard, you cannot get it to work.
https://codesandbox.io/s/codesandbox-00m5y

## Additional information

- Screenshots or code
- Notes

**Comment by tw15egan** [comment]
Since you add `onClick` handler, you should just be able to add `onKeyDown={e => handleClick(e)}` to the element to get it to trigger on keypress. There are no built-in click or key handlers on the `Tag` element

**Comment by emyarod** [comment]
I think before we decide if we should handle this by default, we need some more clarification on usage guidelines for this tag variant

**Comment by abbeyhrt** [comment]
I'm going to close this since it seems the problem has been addressed. See the guidance offered by @tw15egan for the solution.
